### ISSUE TITLE
Investigate Incorrect Anonymization Tag for NIE

### ISSUE THREAD

**Issue by Jmnobles1** [NCATComp410/comp410_fall_2024#73]
ID: 23

In the anonymization output, this was found:

Original: The NIE ######### was anonymized incorrectly. The correct number is #########
Anonymized: The NIE <SG_NRIC_FIN> was incorrect.  The correct number is <ES_NIE>

Expected Behavior:
All Spanish NIE numbers should be anonymized consistently with the <ES_NIE> tag and no other tag. So outputting a tag for the incorrect NIE with <SG_NRIC_FIN> is a issue and should be looked into.

**Comment by claesmk** [comment]
Great use of ### to protect PII in the issue description

**Comment by Pojohnson-XJ** [comment]
I agree with the tag being incorrect it can cause a issue for those coming to look after you and may cause confusion in the future
